File: android/app/src/main/kotlin/com/quicky/ridebahamas/PickerMapNativeFactory.java
package com.quicky.ridebahamas;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import io.flutter.plugin.common.BinaryMessenger;
import io.flutter.plugin.common.StandardMessageCodec;
import io.flutter.plugin.platform.PlatformView;
import io.flutter.plugin.platform.PlatformViewFactory;

import java.util.Map;

public class PickerMapNativeFactory extends PlatformViewFactory {
  private final BinaryMessenger messenger;

  public PickerMapNativeFactory(BinaryMessenger messenger) {
    super(StandardMessageCodec.INSTANCE);
    this.messenger = messenger;
  }

  @Override
  public PlatformView create(@NonNull Context context, int viewId, @Nullable Object args) {
    @SuppressWarnings("unchecked")
    Map<String, Object> creationParams = args instanceof Map ? (Map<String, Object>) args : null;
    return new PickerMapNativeView(context, messenger, viewId, creationParams);
  }
}

File: android/app/src/main/kotlin/com/quicky/ridebahamas/PickerMapNativeFactory.kt
package com.quicky.ridebahamas

import android.content.Context
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

class PickerMapNativeFactory(private val messenger: BinaryMessenger) :
  PlatformViewFactory(StandardMessageCodec.INSTANCE) {
  override fun create(context: Context, id: Int, obj: Any?): PlatformView {
    return PickerMapNativeView(context, messenger, id)
  }
}
